docs/spec/v1/buckets.md

```diff
@@ -651,6 +651,42 @@ When a Bucket's `.spec.provider` is set to `gcp`, the source-controller will
 attempt to communicate with the specified [Endpoint](#endpoint) using the
 [Google Client SDK](https://github.com/googleapis/google-api-go-client).
 
+The `gcp` provider can be used to authenticate automatically using OAuth scopes
+or Workload Identity, and by extension gain access to Google Cloud Storage.
+
+When the GKE nodes have the appropriate OAuth scope for accessing Google Cloud
+Storage, source-controller running on it will also have access to it.
+
+When using Workload Identity to enable access to Google Cloud Storage, add
+the following patch to your bootstrap repository, in the
+`flux-system/kustomization.yaml` file:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - gotk-components.yaml
+  - gotk-sync.yaml
+patches:
+  - patch: |
+      apiVersion: v1
+      kind: ServiceAccount
+      metadata:
+        name: source-controller
+        annotations:
+          iam.gke.io/gcp-service-account: <identity-name>
+    target:
+      kind: ServiceAccount
+      name: source-controller
+```
+
+The Google Cloud Storage service uses the permission `storage.objects.get`
```

Comment:
We can just point here: https://fluxcd.io/flux/integrations/gcp/#for-google-cloud-storage